src/client/src/halcs_client_core.c

```diff
@@ -4228,48 +4228,61 @@ PARAM_FUNC_CLIENT_READ(rtmlamp_ohwr_pi_sp_lim_inf)
 }
 
 /**************** FOFB PROCESSING SMIO Functions ****************/
+halcs_client_err_e halcs_fofb_processing_coeff_ram_bank_read(
+    halcs_client_t *self, char *service, const uint32_t chan,
+    smio_fofb_processing_data_block_t *const coeffs) {
+    assert(self);
+    assert(service);
 
-/* FOFB processing ram write enable */
-PARAM_FUNC_CLIENT_WRITE(fofb_processing_ram_write)
-{
-    return param_client_write (self, service, FOFB_PROCESSING_OPCODE_SET_GET_RAM_WRITE, fofb_processing_ram_write);
-}
+    uint32_t input;
+    uint32_t output[FOFB_PROCESSING_REGS_RAM_BANK_SIZE/sizeof(uint32_t)];
 
-PARAM_FUNC_CLIENT_READ(fofb_processing_ram_write)
-{
-    return param_client_read (self, service, FOFB_PROCESSING_OPCODE_SET_GET_RAM_WRITE, fofb_processing_ram_write);
-}
+    input = chan;
 
-/* FOFB processing ram addr */
-PARAM_FUNC_CLIENT_WRITE(fofb_processing_ram_addr)
-{
-    return param_client_write (self, service, FOFB_PROCESSING_OPCODE_SET_GET_RAM_ADDR, fofb_processing_ram_addr);
-}
+    halcs_client_err_e err = halcs_func_trans_exec(self,
+        FOFB_PROCESSING_NAME_COEFF_RAM_BANK_READ, service, &input, output);
```

Comment:
```suggestion
        FOFB_PROCESSING_NAME_COEFF_RAM_BANK_READ, service, &input, coeffs->data);
```

Para evitar a cópia?